torch/csrc/autograd/init.cpp

```diff
@@ -202,20 +211,22 @@ PyObject* THPAutograd_initExtension(PyObject* _unused, PyObject *unused) {
   py::class_<ProfilerResult>(m, "_ProfilerResult")
     .def("trace_start_us", &ProfilerResult::trace_start_us)
     .def("events", &ProfilerResult::events)
-    .def("save", &ProfilerResult::save);
+#ifdef USE_KINETO
+    .def("save", &ProfilerResult::save)
+#endif // USE_KINETO
+    ;
 
   m.def("_enable_profiler", enableProfiler);
   m.def("_disable_profiler", disableProfiler);
   m.def("_prepare_profiler", prepareProfiler);
-#endif
 
   m.def("_add_metadata_json", [](const std::string& key, const std::string& value) {
 #ifdef USE_KINETO
       addMetadataJson(key, value);
```

Comment:
Interesting. I did not realize that this function was available only with kineto.

Reply:
yes, that's to save the extra data into the kineto events, to be saved later in the trace (by kineto); we don't support this in the non-kineto case; generally speaking, profiler without kineto has less capabilities, and that's expected; in almost all use cases profiler is built with kineto support, I'd recommend exploring this for the remaining cases as, e.g. mobile builds - cpu-only kineto library should not add too much to the build size

Reply:
I hope that size burden is not too much. But profiling build wont be production build so we wont include kineto dependency in production anyway, but it would still be nice to optionally include cpu only kineto lib, or at least for internal builds we can have separate cpu only target to link against.